### ISSUE TITLE
Custom service

### ISSUE THREAD

**Issue by AyWa** [zalando-incubator/es-operator#113]
## Question

I have an overall question about `es-operator`. I am migrating from the official ES helm chart (for the data nodes the other node are still deploy with helm). However I am stuck to the last part: `configuring my ingress`

On the official helm chart, I can setup an ingress / services like:

```
ingress:
  enabled: True
  annotations:
      kubernetes.io/ingress.class: alb
      alb.ingress.kubernetes.io/target-type: instance
      alb.ingress.kubernetes.io/subnets: subnet-XX,subnet-YY
      alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 9200}]'
      alb.ingress.kubernetes.io/healthcheck-path: '/_cluster/health'
  hosts: [""]
  path: "/*"
  tls: []

service:
  type: LoadBalancer
``` 
Which will create 2 services (one load balancer and one headless)

However I can not (or find a way) to setup this load balancer correctly. From what I saw from the code, the operator need to own the service and will create a NodePort service. 

Does there is anyway to solve that ?

What I will do in the meantime is deploy some coordinate nodes with the load balancer, so I can access the Elasticsearch cluster (or do that on the master node for testing)

**Comment by mikkeloscar** [comment]
The es-operator creates a service for each ElasticsearchDataSet (eds) that you define. This is mostly intended for the operator to communicate with Elasticsearch API, but ofc you can also use it for ingress (I think). We currently don't allow you to e.g. change this to a `type: LoadBalancer` service, but you could just deploy your own service which targets the EDS with the same label selectors.

Your config snippet suggest that you both create an ALB for ingress AND an ELB for service type LoadBalancer. I don't know if this is intentional or what purpose it serves, but if an ingress is enough then you can create an ingress resource that points to the service created by es-operator.

I hope this clarifies it a bit.

**Comment by AyWa** [comment]
First of all, thank you a lot for your quick response 👍 

So far I was trying to overwrite the service and creating the ingress ( add my own label selectors), but knowing that I can not change `type: LoadBalancer` I will then try to follow your advice:

- deploy normally the `EDS` 
- deploy my ingress targeting a `NodePort` service with the existing label selectors


My previous configuration was creating an `ALB` for the ingress. And the Ingress was targeting the `load balancer` service.


Anyway, I will comment tomorrow if I succeed or not to deploy my `ingress`.

**Comment by AyWa** [comment]
I think we can close that. I succeed to create my ingress and a service that target the `EDS`

If I can suggest something, is that the annotations defined in the `EDS` should be set on the stateful set.
In fact my issue was coming from that. (I did not realize at first but `kubectl get pods -n es --show-labels` help me to realize it). 

So so far, I use the `application=elasticsearch` to target the `EDS`
